utils/isOwner.js

```diff
@@ -0,0 +1,20 @@
+const isOwner = async (collection, userId, id) => {
+  try {
+    const singleCollection = await collection.findOne({ id });
+
+    if(!singleCollection) throw new Error('A record with that id was not found');
+
+    if(!singleCollection.user?.id) throw new Error('This collection does not have a user id');
+
+    if(userId === singleCollection.user.id){
+      return true;
+    }
+
+    return false;
+  } catch(err) {
```

Comment:
The error you throw in the try branch will be caught here. Just want to make sure this is what you intended.

Reply:
Ah I thought it would be caught in the catch block where I called the function

Reply:
My original idea was that those errors would flow on top and be caught in the catch block and then I could return a HTTP error message back to the caller of the API

Reply:
Did some googling and figured out I could bubble up the error if I rethrow the exception of the catch block in the isOwner function. The only thing is I won't be able to return false since throw exists the function.

Reply:
I think it's better to return false and let the caller decides the HTTP error. There is a chance that the caller doesn't catch the error if this function returns an error.

Reply:
so for the two checks on lines 5 and 7 I should return false instead? How would they differentiate from the error?